File: src/test/java/com/ject/studytrip/member/fixture/UpdateMemberRequestFixture.java
package com.ject.studytrip.member.fixture;

import com.ject.studytrip.member.presentation.dto.request.UpdateMemberRequest;

public class UpdateMemberRequestFixture {
    private String nickname = null;
    private String category = null;

    public UpdateMemberRequestFixture withNickname(String nickname) {
        this.nickname = nickname;
        return this;
    }

    public UpdateMemberRequestFixture withCategory(String category) {
        this.category = category;
        return this;
    }

    public UpdateMemberRequest build() {
        return new UpdateMemberRequest(nickname, category);
    }
}

File: src/test/kotlin/com/ject/studytrip/member/fixture/UpdateMemberRequestFixture.kt
package com.ject.studytrip.member.fixture

import com.ject.studytrip.member.presentation.dto.request.UpdateMemberRequest

class UpdateMemberRequestFixture(
    private val nickname: String? = null,
    private val category: String? = null,
) {
    fun withNickname(nickname: String?): UpdateMemberRequestFixture = UpdateMemberRequestFixture(nickname, category)

    fun withCategory(category: String?): UpdateMemberRequestFixture = UpdateMemberRequestFixture(nickname, category)

    fun build(): UpdateMemberRequest = UpdateMemberRequest(nickname, category)
}
